tests/functional/test_systemstatus.py

```diff
@@ -8,9 +8,9 @@
 import os
 
 # Functional tests for system status as a basic connectivity/anchore-engine up test
-anchore_user = os.environ.get('ANCHORE_CLI_USER', 'admin')
```

Comment:
why, remove the defaults?

Reply:
When moving the test harness to a containerized setup (the `job-runner` container in the compose stack), default values were hiding in a few different places, and were non-obvious to find and override. This was an effort to move all defaults to be derived from a single source of truth, failing fast if the value is missing.

For example, the `os.environ.get('ANCHORE_CLI_URL', 'http://localhost:8228/v1')` line works fine when running on localhost or if the container is forwarding ports to the host, but from container-to-container the value needs to change. There isn't a "hint" to a user of the functional test of what values need to be set from a compose file or otherwise --removing the default makes this obvious by forcing a single source of truth, the environment variable.

This may make local dev harder, in which case, we should think about a good compromise --maybe put back some defaults, but place them in a single spot in the functional codebase as a chokepoint.

Reply:
After reading this comment thread and discussing it a couple of days ago I am going to push back on this change. This line is not _changing values_, it is assigning them if they are not set. 

I agree that it is non-obvious that there are tests like this one relying on environment variables, which aren't advertised  clearly. This could be changed to be put into a central location where environment variables are set. In my case, I tend to expect them in a `conftest.py`.

I also agree that there is no "hint" of what values are being used. This is another thing that can be improved with Pytest itself, where the report header can be enhanced to add the environment variables that are crucial for the test.

The single source of truth for the values is an excellent approach, and once again I agree that it is worth pursuing. However, a single source of truth should _not force_ a user to have to source a file before running tests. The addition of a file and the requirement of sourcing it is a step back for encouraging local development as it creates a hassle.

I really like the idea of the compromise suggested by @wagoodman here. I can follow up with a PR that addresses these concerns:

1. Put the defaults back as a fallback only
1. All environment variables should go into a central location - `tests/functcional/conftest.py` is a good spot for these
1. Add a Pytest hook to enhance the report header to include the variables (and their source)

This comment should not prevent this PR from moving forward